app/resources/inventory.py

```diff
@@ -6,6 +6,16 @@
 from vars import game_info
 
 
+@m.user_endpoint(path=["inventory", "trade"], requires=trade_requirements)
+def trade(data: dict, user: str) -> dict:
+    element: Optional[Inventory] = wrapper.session.query(Inventory).filter_by(element_uuid=data["element_uuid"]).first()
+    if element is None or element.owner != user or element.owner == data["target"]:
```

Comment:
Return a different error message if the user is not the owner of the item or if the target user already owns the item. Also use the error messages defined in `schemes.py` and create new ones if necessary.

Reply:
Yes. You can also improve the readability by checking:
`user == data["target"]`
instead of
`element.owner == data["target"]`